ktor-http/ktor-http-cio/src/io/ktor/http/cio/internals/WeakTimeoutQueue.kt

```diff
@@ -74,34 +74,29 @@ class WeakTimeoutQueue(
     suspend fun <T> withTimeout(block: suspend CoroutineScope.() -> T): T {
         return suspendCoroutineUninterceptedOrReturn { rawContinuation ->
             val continuation = rawContinuation.intercepted()
+
             val wrapped = WeakTimeoutCoroutine(continuation.context, continuation)
             val handle = register(wrapped)
-            wrapped.disposeOnCompletion(handle)
+            wrapped.invokeOnCompletion(handle)
```

Comment:
If `withTimeout` provides a scope, then it should wait for all children (and aggregate their exceptions if there are some) before exiting that scope.
E.g.
```
q.withTimeout {
                launch(start = CoroutineStart.ATOMIC) {
                    Thread.sleep(2000)
                    println("Child finished")
                }
                throw MyException()
            }
```

what is expected behaviour here?
In kx.coroutines there it will wait for the child before throwing exception